src/openqaoa-core/openqaoa/algorithms/qaoa/qaoa_result.py

```diff
@@ -629,3 +629,56 @@ def lowest_cost_bitstrings(self, n_bitstrings: int = 1) -> dict:
             ],
         }
         return best_results
+    
+    def calculate_statistics(self, include_intermediate=False) -> dict:
+        """
+        A function to calculate statistics of measurement outcomes associated with a QAOA workflow
+
+        Parameters
+        ----------
+        include_intermediate: `bool`
+            Whether it is necessary to calculate statistics for intermediate results. Defaults to False.
+        """
+        if len(self.intermediate["measurement_outcomes"]) == 0 and include_intermediate == True:
+            raise ValueError(
+                "Please specify saving intermediate measurements during optimization."
```

Comment:
Maybe we can be more explicit. How about:
 
"_The underlying QAOA object does not seem to have any intermediate measurement result. Please, consider saving intermediate measurements during optimization by setting `optimization_progress=True` in your workflow._"

Or a less verbose version of it: the important thing (I think) is to give the end user a pointer to what they need to do in order to use this feature :)